### PR TITLE
Android 5.2.0 Adaption & Kotlin migration & Gradle bumping

### LINKED ISSUE #145: Bump Android native SDK `5.2.0`
Naver Android SDK `5.2.0` 엔 현재 production에서도 이슈가 되는 것들의 몇가지 fix가 포함되어있지만 `authenticate` 함수가 뜬금없이 `ActivityResultLauncher` 를 요구해서 내부적인 구현이 어려워졌습니다.

### PR DESCRIPTION
**Pull request by mym0404**
close #145

#### This should be a breaking change because of `5.2.0` SDK api change.

But I [issued](https://github.com/naver/naveridlogin-sdk-android/issues/55) about this. 

## Works
- Bump `minSdkVersion` `16` -> `21` (Naver SDK minimum requirement)
- Bump `compileSdkVersion` `28` -> `30`
- Bump `targetSdkVersion` `28` -> `30`
- Bump `buildToolsVersion` `28.x` -> `30.0.2`
- Bump `naverSDK` `5.1.0` -> `5.2.0`
- Refactor Gradle
- Remove Manifest contents [CHANGELOG](https://github.com/naver/naveridlogin-sdk-android/wiki/v5.0.0-%EB%AF%B8%EB%A7%8C-%EA%B0%80%EC%9D%B4%EB%93%9C#101-sdk-41x-%EB%B2%84%EC%A0%84%EC%97%90%EC%84%9C-420-%EB%B2%84%EC%A0%84%EC%9C%BC%EB%A1%9C-%EC%97%85%EA%B7%B8%EB%A0%88%EC%9D%B4%EB%93%9C-%EB%B0%A9%EB%B2%95)

Sdk Activities are included well after the Manifest file is removed.
The following screenshot is the result of Manifest merging after shrinking Manifest file.
image

- Integrate Kotlin into Gradle configuration
- Migrate code to Kotlin (1.7.10 default version) + Refactoring

---
## Test
- In my local environment, it works with a local yarn package with path installation. 
---
## Caveats
- I didn't convert package.java to Kotlin because of RN deprecated `createJsModule` override issue
- I can't be sure the global object in `RNNaverLoginModule.kt` works well without memory leak or other sub-problems.
---
## Trial I need your QAs
```json
"@react-native-seoul/naver-login": "mym0404/react-native-naver-login#9da4c97d8de58ce941d041f77a262769187daf0d"
```

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -1,41 +1,48 @@
...
   }
+  compileOptions {
+    sourceCompatibility JavaVersion.VERSION_1_8
+    targetCompatibility JavaVersion.VERSION_1_8
+  }
 }
 
 repositories {
   mavenCentral()
+  google()
```
Comment (mym0404) [inline android/build.gradle:41]: 없어도 될듯합니다.

**Comment by mym0404** [comment]
@hyochan 네 시간나면 작업하도록 하겠습니다.

**Comment by hyochan** [comment]
> @hyochan 네 시간나면 작업하도록 하겠습니다.

빠른 답변 감사드립니다. 메인터네이너 권한 드리는 것도 진행해도 될까요?

**Comment by mym0404** [comment]
네 진행해주세요 @hyochan

**Comment by hyochan** [comment]
> 네 진행해주세요 @hyochan

진행했습니다~ 감사합니다

**Comment by mym0404** [comment]
`3.x` 에서 같이 merge 될 것입니다.

**Pull request closed without merging**